.github/workflows/lint.yml

```diff
@@ -0,0 +1,61 @@
+name: Lint
+
+on:  [push, pull_request]
+
+env:
+  PLUGIN_NAME: dashboards-observability
+  OPENSEARCH_DASHBOARDS_VERSION: 'main'
+
+jobs:
+  build:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout OpenSearch Dashboards
+        uses: actions/checkout@v2
+        with:
+          repository: opensearch-project/Opensearch-Dashboards
+          ref: ${{ env.OPENSEARCH_DASHBOARDS_VERSION }}
+          path: OpenSearch-Dashboards
+
+      - name: Checkout dashboards observability
+        uses: actions/checkout@v2
+        with:
+          path: OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}         
+
+      - name: Get node and yarn versions
+        working-directory: ${{ env.WORKING_DIR }}
+        id: versions_step
+        run: |
+          echo "::set-output name=node_version::$(cat ./OpenSearch-Dashboards/.nvmrc | cut -d"." -f1)"
+          echo "::set-output name=yarn_version::$(node -p "(require('./OpenSearch-Dashboards/package.json').engines.yarn).match(/[.0-9]+/)[0]")"
+
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ steps.versions_step.outputs.node_version }}
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install correct yarn version for OpenSearch Dashboards
+        run: |
+          npm uninstall -g yarn
+          echo "Installing yarn ${{ steps.versions_step.outputs.yarn_version }}"
+          npm i -g yarn@${{ steps.versions_step.outputs.yarn_version }}
+
+      - name: Bootstrap the plugin
+        working-directory: OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}  
+        run:
+          yarn osd bootstrap
+
+      - name: lint code base
+        working-directory: OpenSearch-Dashboards/plugins/${{ env.PLUGIN_NAME }}
+        run: |
+          git fetch origin main
+          CHANGED_FILES=$(git diff --name-only --diff-filter=ACMRTUXB origin/main | grep -E "\.(js|ts|tsx)$")
+          if [ -n "$CHANGED_FILES" ]; then
+            echo "Linting changed files..."
+            yarn lint $CHANGED_FILES
```

Comment:
why not combine with the unit test CI?

also probably not an issue but i believe files with spaces won't work

Reply:
I don't think it cannot be combined with unit test CI, and saw other plugin has this integrated with their unit tests CI, so it should be fine. The initial considerations are

- In our case since we have tech debt for these linting errors, and no enforced rule so far to ensure anyone who make changes to any file, they are responsible for fixing all existing linting errors in the file(s) they modify. Then we may want to skip failing lintings in sometime, but if it's merged with unit test CI, the following Unit test run will be blocked in those cases then we cannot verify what are results of unit test running.
- other reasons like they are doing different checks, I'm just thinking conceptually we should separate them maybe?
- And if we want to do granularly controls like not runing linting on push but only on pull_request, may take additional effort to do them differently in one CI maybe

Reply:
Yes, spaces should not be an issue I think. Will we have (or allow) file(s) with space in the names?

Reply:
i see.

> spaces should not be an issue I think

i raised up since it might cause a bug and can be fixed using a bash array, but if we don't have space in filenames then i'm ok with it